sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -114,45 +114,49 @@ export class AzureCliCredential implements TokenCredential {
     ensureValidScope(scope, logger);
     const resource = getScopeResource(scope);
 
-    let responseData = "";
-
     return tracingClient.withSpan(`${this.constructor.name}.getToken`, options, async () => {
       try {
         const obj = await cliCredentialInternals.getAzureCliAccessToken(resource, tenantId);
-        if (obj.stderr) {
-          const isLoginError = obj.stderr.match("(.*)az login(.*)");
-          const isNotInstallError =
-            obj.stderr.match("az:(.*)not found") || obj.stderr.startsWith("'az' is not recognized");
-          if (isNotInstallError) {
-            const error = new CredentialUnavailableError(
-              "Azure CLI could not be found.  Please visit https://aka.ms/azure-cli for installation instructions and then, once installed, authenticate to your Azure account using 'az login'."
-            );
-            logger.getToken.info(formatError(scopes, error));
-            throw error;
-          } else if (isLoginError) {
-            const error = new CredentialUnavailableError(
-              "Please run 'az login' from a command prompt to authenticate before using this credential."
-            );
-            logger.getToken.info(formatError(scopes, error));
-            throw error;
-          }
-          const error = new CredentialUnavailableError(obj.stderr);
+        const isLoginError = obj.stderr?.match("(.*)az login(.*)");
```

Comment:
curious what's the value of `obj.error` when there's only warning and whether it can be used to determine whether it's an error case.

Reply:
We would need to have examples of relevant warnings 🤔

Reply:
@jeremymeng Based on this documentation - https://nodejs.org/api/child_process.html#child_processexecfilefile-args-options-callback they have this information -
```
If this method is invoked as its [util.promisify()](https://nodejs.org/api/util.html#utilpromisifyoriginal)ed version, it returns a Promise for an Object with stdout and stderr properties. The returned ChildProcess instance is attached to the Promise as a child property. In case of an error (including any error resulting in an exit code other than 0), a rejected promise is returned, with the same error object given in the callback, but with two additional properties stdout and stderr.
```
This implies that for us, the `stderr` and `error` have the same content. If there is a scenario where the `error` filed has additional details not present in `stderr` then we can account for it in the code. Otherwise, it is not necessary.